pkgs/applications/misc/keymapp/default.nix

```diff
@@ -0,0 +1,68 @@
+{ stdenv
+, lib
+, fetchurl
+, autoPatchelfHook
+, wrapGAppsHook
+, libusb1
+, webkitgtk
+, gtk3
+, writeShellScript
+, makeDesktopItem
+, copyDesktopItems
+}:
+let
+  desktopItem = makeDesktopItem {
+    name = "keymapp";
+    icon = "keymapp";
+    desktopName = "Keymapp";
+    categories = [ "Settings" "HardwareSettings" ];
+    type = "Application";
+    exec = "keymapp";
+  };
+in
+stdenv.mkDerivation rec {
+  pname = "keymapp";
+  version = "1.0.5";
+
+  src = fetchurl {
+    url = "https://oryx.nyc3.cdn.digitaloceanspaces.com/keymapp/keymapp-${version}.tar.gz";
+    hash = "sha256-e9Ty3gMb+nkXGK8sND4ljyrIxP+1fLasiV6DoTiWmsU=";
+  };
+
+  nativeBuildInputs = [
+    copyDesktopItems
+    autoPatchelfHook
+    wrapGAppsHook
+  ];
+
+  buildInputs = [
+    libusb1
+    webkitgtk
+    gtk3
+  ];
+
+  sourceRoot = ".";
+
+  installPhase = ''
+    runHook preInstall
+
+    install -m755 -D keymapp "$out/bin/${pname}"
+    install -Dm644 ${./keymapp.png} "$out/share/pixmaps/keymapp.png"
```

Comment:
Can we fetch the image instead?

Reply:
Upstream doesn't provide id. I got it from archlinux package, so we could use that:
https://aur.archlinux.org/cgit/aur.git/plain/keymapp.png?h=zsa-keymapp-bin

Also I can host the image somewhere if that is preffered.

Reply:
Or alternative we could ask ZSA if they can provide a image source?
If you don't like to, i can send them a mail about that.

Reply:
It would be easiest if they could bundle the image in that .tar.gz file they host.

Could you ask them please? You already have a thread about keymapp with them, don't want to start a new one.

Reply:
I have sent them a E-Mail, i wanted anyways to mail them the days, as i was asked for Feedback on my Voyager experience :).  I will report back as soon as i get a respond, I asked if they would either place a desktop icon in the archive or provide a URL where we can fetch the icon form.

Reply:
> [aur.archlinux.org/cgit/aur.git/plain/keymapp.png?h=zsa-keymapp-bin](https://aur.archlinux.org/cgit/aur.git/plain/keymapp.png?h=zsa-keymapp-bin)

We could just fetch it from there 😅 and avoid the next binary blob in the tree

Reply:
So I did already receive a half response, sadly not on the part for the PR. So I would suggest first to fetch the PNG from AUR and I can update the Code, as soon as it get new info (to prevent the PR to get stale)

Reply:
> Hello,
>
>I'm going to add the icon to the tarball starting from next release expected end of this week. It will be called icon.png. Let me know if it works for you ?
>
>Best,
>Florian Didron,
>Lead Developer,
>ZSA Technology Labs Inc.

I think that this should be fine/good for our situation.

Reply:
>Hi,
>
>I just released version 1.0.7 https://oryx.nyc3.cdn.digitaloceanspaces.com/keymapp/keymapp-1.0.7.tar.gz . The tarball includes the icon.png file.
>
>Best,
>Florian Didron,
>Lead Developer,
>ZSA Technology Labs Inc.

I already tried it out on my own package.
Icon is a bit different to what the Archlinux folks are using, but as its upstream i take it! :slightly_smiling_face: 
That is the patch I applied on my own package.

https://github.com/Shawn8901/nix-configuration/commit/f30d343f364028964c0c6c9270c30159f8beb30f

Reply:
Thanks I modified it to use the icon from image and removed the binary blob from history.

It works well. And that should be everything. I think we can ship it now, just in time to make this a good Christmas present.